File: server/src/main/java/de/ingrid/igeserver/model/UserInfo.java
package de.ingrid.igeserver.model;


import java.util.List;
import java.util.Set;

public class UserInfo {
    public String userId;
    public String name;

    public List<Catalog> assignedCatalogs;
    public Set<String> roles;

    // the current selected catalog in case a user can access multiple catalogs
    public Catalog currentCatalog;
}

File: server/src/main/java/de/ingrid/igeserver/persistence/postgresql/jpa/model/UserInfo.kt
package de.ingrid.igeserver.persistence.postgresql.jpa.model

import de.ingrid.igeserver.annotations.NoArgs
import de.ingrid.igeserver.persistence.postgresql.jpa.EntityBase
import javax.persistence.*

@NoArgs
@Entity
@Table(name = "user_info")
class UserInfo(
    @Column(nullable = false)
    val userId: String,

    @ManyToOne
    @JoinColumn(name="cur_catalog_id", nullable=false)
    val curCatalog: Catalog
) : EntityBase() {

    @ManyToMany(mappedBy = "users")
    var catalogs: Set<Catalog> = setOf()
}